math/power_of_two.cpp

```diff
@@ -0,0 +1,58 @@
+
+/**
+ * @file power_of_two.cpp
+ * @brief Implementation to check whether a number is a power of 2 or not.
+ *
+ * @details
+ * This algorithm uses bit manipulation to check if a number is a power of 2 or not.
+ *
+ * ### Algorithm
+ * Let the input number be n, then the bitwise and between n and n-1 will let us know 
+ * whether the number is power of 2 or not
+ * 
+ * For Example,
+ * If N= 32 then N-1 is 31, if we perform bitwise and of these two numbers then
+ * the result will be zero, which indicates that it is the power of 2
+ * If N=23 then N-1 is 22, if we perform bitwise and of these two numbers then
+ * the result will not be zero , which indicates that it is not the power of 2
+ * \note This implementation is better than naive recursive or iterative approach. 
+ * 
+ * @author [Neha Hasija](https://github.com/neha-hasija17)
+ */
+ 
+#include <iostream>
+/**
+ * Function to test above algorithm
```

Comment:
Missing parameter documentation.
```suggestion
 * @brief Function to test above algorithm
 * @param n description
 * @returns void
```